File: list-models.py
import urllib.request
import json
import re

with open('.env', 'r') as f:
    env_content = f.read()

match = re.search(r'VITE_GEMINI_API_KEY=(.*)', env_content)
if not match:
    print("No key found")
    exit(1)

key = match.group(1).strip()
url = f"https://generativelanguage.googleapis.com/v1beta/models?key={key}"

try:
    req = urllib.request.urlopen(url)
    data = json.loads(req.read().decode())
    print(json.dumps([m['name'] for m in data.get('models', [])], indent=2))
except Exception as e:
    print(e)

File: list-models.js
import fs from 'fs';

async function run() {
  try {
    const envFile = fs.readFileSync('.env', 'utf8');
    const keyMatch = envFile.match(/VITE_GEMINI_API_KEY=(.*)/);
    const key = keyMatch ? keyMatch[1].trim() : '';

    const req = await fetch(`https://generativelanguage.googleapis.com/v1beta/models?key=${key}`);
    const data = await req.json();
    console.log(JSON.stringify(data.models.map(m => m.name), null, 2));
  } catch (e) {
    console.error(e);
  }
}
run();
